# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/jit/AliasAnalysis.cpp
# Commit: 229759a67f4f
# Full Hash: 229759a67f4f26ccde9f7bde5423cfd82b216fa2
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2019-03-22 01:23:00
# Description:
#   Bug 1537924 - Simplify some alias sets in Ion. r=tcampbell, a=dveditz
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D24400
# ==============================================================================

diff -r feda786b35cb -r 229759a67f4f js/src/jit/AliasAnalysis.cpp
--- a/js/src/jit/AliasAnalysis.cpp	Thu Mar 21 23:59:53 2019 +0200
+++ b/js/src/jit/AliasAnalysis.cpp	Thu Mar 21 17:52:21 2019 +0000
@@ -133,7 +133,6 @@
     case MDefinition::Opcode::SetDisjointTypedElements:
     case MDefinition::Opcode::ArrayPopShift:
     case MDefinition::Opcode::ArrayPush:
-    case MDefinition::Opcode::ArraySlice:
     case MDefinition::Opcode::LoadTypedArrayElementHole:
     case MDefinition::Opcode::StoreTypedArrayElementHole:
     case MDefinition::Opcode::LoadFixedSlot:
@@ -158,6 +157,7 @@
       object = ins->getOperand(0);
       break;
     case MDefinition::Opcode::GetPropertyCache:
+    case MDefinition::Opcode::CallGetProperty:
     case MDefinition::Opcode::GetDOMProperty:
     case MDefinition::Opcode::GetDOMMember:
     case MDefinition::Opcode::Call:
@@ -184,6 +184,7 @@
     case MDefinition::Opcode::WasmLoadRef:
     case MDefinition::Opcode::WasmStoreRef:
     case MDefinition::Opcode::ArrayJoin:
+    case MDefinition::Opcode::ArraySlice:
       return nullptr;
     default:
 #ifdef DEBUG